solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,530 @@
+    using SafeMath for uint256;
+    using Math for uint256;
+    using SafeERC20 for IERC20;
+
+    struct Strategy {
+        address[][] paths;
+        uint256[] amounts;
+        address[] govPath;
+    }
+
+    struct NetNetworkTokenConversionAmounts {
+        uint256 amount;
+        uint256 incentiveFeeAmount;
+    }
+
+    // the vortex is only designed to work with 50/50 standard pool converters
+    uint32 private constant STANDARD_POOL_RESERVE_WEIGHT = PPM_RESOLUTION / 2;
+
+    // the type of the standard pool converter
+    uint16 private constant STANDARD_POOL_CONVERTER_TYPE = 3;
+
+    // the address of the network token
+    IERC20 private immutable _networkToken;
+
+    // the address of the governance token
+    IERC20 private immutable _govToken;
+
+    // the address of the governance token security module
+    ITokenGovernance public immutable _govTokenGovernance;
+
+    // the percentage of the converted network tokens to be sent to the caller of the vortex (in units of PPM)
+    uint32 private _burnIncentiveFee;
+
+    // the maximum incentive fee to be sent to the caller of the vortex
+    uint256 private _maxBurnIncentiveFeeAmount;
+
+    // stores the total amount of the burned governance tokens
+    uint256 private _totalBurnedAmount;
+
+    /**
+     * @dev triggered when the burn incentive fee has been changed
+     *
+     * @param prevBurnIncentiveFee the previous burn incentive fee (in units of PPM)
+     * @param newBurnIncentiveFee the new burn incentive fee (in units of PPM)
+     * @param prevMaxBurnIncentiveFeeAmount the previous maximum burn incentive fee
+     * @param newMaxBurnIncentiveFeeAmount the new maximum burn incentive fee
+     */
+    event BurnIncentiveFeeUpdated(
+        uint32 prevBurnIncentiveFee,
+        uint32 newBurnIncentiveFee,
+        uint256 prevMaxBurnIncentiveFeeAmount,
+        uint256 newMaxBurnIncentiveFeeAmount
+    );
+
+    /**
+     * @dev triggered after a completed vortex
+     *
+     * @param tokens the converted tokens
+     * @param amounts the amounts of the converted tokens
+     * @param conversionAmounts the network token amounts the tokens were converted to
+     * @param totalBurnedAmount the total burned amount in this vortex
+     */
+    event Burned(IERC20[] tokens, uint256[] amounts, uint256[] conversionAmounts, uint256 totalBurnedAmount);
+
+    /**
+     * @dev initializes a new VortexContract contract
+     *
+     * @param networkToken the address of the network token
+     * @param govTokenGovernance the address of the governance token security module
+     * @param burnIncentiveFee the percentage of the converted network tokens to be sent to the caller of the vortex (in units of PPM)
+     * @param maxBurnIncentiveFeeAmount the maximum incentive fee to be sent to the caller of the vortex
+     * @param registry the address of the contract registry
+     */
+    constructor(
```

Comment:
I'd separate the settings from the construction

Reply:
👍